src/sagemaker/jumpstart/utils.py

```diff
@@ -214,31 +209,33 @@ def verify_model_region_and_return_specs(
             f"JumpStart model ID '{model_id}' and version '{version}' " "does not support training."
         )
 
-    if model_specs.deprecated and not tolerate_deprecated_model:
-        raise DeprecatedJumpStartModelError(model_id=model_id, version=version)
-
-    if (
-        scope == constants.JumpStartScriptScope.INFERENCE.value
-        and model_specs.inference_vulnerable
-        and not tolerate_vulnerable_model
-    ):
-        raise VulnerableJumpStartModelError(
-            model_id=model_id,
-            version=version,
-            vulnerabilities=model_specs.inference_vulnerabilities,
-            scope=constants.JumpStartScriptScope.INFERENCE,
+    if model_specs.deprecated:
+        if not tolerate_deprecated_model:
+            raise DeprecatedJumpStartModelError(model_id=model_id, version=version)
+        LOGGER.warning("Using deprecated JumpStart model '%s' and version '%s'.", model_id, version)
+
+    if scope == constants.JumpStartScriptScope.INFERENCE.value and model_specs.inference_vulnerable:
+        if not tolerate_vulnerable_model:
+            raise VulnerableJumpStartModelError(
+                model_id=model_id,
+                version=version,
+                vulnerabilities=model_specs.inference_vulnerabilities,
+                scope=constants.JumpStartScriptScope.INFERENCE,
+            )
+        LOGGER.warning(
+            "Using vulnerable JumpStart model '%s' and version '%s' (inference).", model_id, version
```

Comment:
non-blocking: while I agree that logging a warning is good practice, there is one downside to this approach. The typical use case of our utilities is:

```py
model_id = "pytorch-ic-mobilenet-v2"
version = "1.1.2"

train_image = image_uris.retrieve(model_id=model_id, version=version, tolerate_deprecated_model=True)
train_model = model_uris.retrieve(model_id=model_id, version=version, tolerate_deprecated_model=True)
train_script = script_uris.retrieve(model_id=model_id, version=version, tolerate_deprecated_model=True)

estimator = Estimator(image_uri=train_image, source_dir=train_script, model_uri=train_model)
[...]
```

In that case, we would log 3 times the same warning, correct?

This may be the correct tradeoff, we can discuss. One possibility would be to add another state-holding class like [SageMakerSetting](https://github.com/aws/sagemaker-python-sdk/blob/d9d8c68b4bfba9b1adf1d8356de092331d70039b/src/sagemaker/jumpstart/accessors.py#L21) to handle such warnings. The class could store some metadata of the warnings (eg model+version) used and associate a time stamp, and bypass the warning if one was raised less than 10 minutes ago, for example.

In any case, this case be addressed in a separate PR.

Reply:
I don't think it's worth it right now to engineer a solution to this problem. After all, most people will not be using deprecated/vulnerable models, and no models right now have this flag.